### PR TITLE
Feature/hatoba cluster

### PR DESCRIPTION
**Pull request by aokumasan**
## Description

- Add nifcloud_hatoba_cluster esource.

## How Has This Been Tested?

- [x]  Buld with `make install` command, and Apply example.

```:console
make install
cd examples/hatoba_cluster
terraform init -plugin-dir ~/.terraform.d/plugins
terraform plan
terraform apply
```

- [x] Run acceptance tests.

```:console
TF_ACC=1 go test ./nifcloud/acc/... -v -count 1 -parallel 20 -timeout 360m -run TestAcc_HatobaCluster
```

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

### CODE REVIEW

File: nifcloud/resources/hatoba/cluster/schema.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			Default: schema.DefaultTimeout(5 * time.Minute),
+			Create:  schema.DefaultTimeout(120 * time.Minute),
+			Update:  schema.DefaultTimeout(120 * time.Minute),
+			Delete:  schema.DefaultTimeout(60 * time.Minute),
+		},
+	}
+}
+
+func newSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
```
[the remark this answers is not in the document]
Comment (aokumasan) [reply nifcloud/resources/hatoba/cluster/schema.go:36]: Sorry 🙏 Fixed!

File: nifcloud/resources/hatoba/cluster/schema.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"node_pools": {
+			Type:     schema.TypeSet,
+			Required: true,
+			MinItems: 1,
+			Set: func(v interface{}) int {
+				name := v.(map[string]interface{})["name"].(string)
```
[the remark this answers is not in the document]
Comment (aokumasan) [reply nifcloud/resources/hatoba/cluster/schema.go]: 😇 

I fixed node pool update logic at 48705919ceba344ad60ca3fbb0ccb4a6b5fc8303

File: nifcloud/resources/hatoba/cluster/update.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	if d.HasChanges("name", "description", "kubernetes_version", "addons_config") {
+		input := expandUpdateClusterInput(d)
+		req := svc.UpdateClusterRequest(input)
+		if _, err := req.Send(ctx); err != nil {
+			return diag.FromErr(fmt.Errorf("failed updating Hatoba cluster: %s", err))
+		}
+
+		d.SetId(d.Get("name").(string))
```
[the remark this answers is not in the document]
Comment (aokumasan) [reply nifcloud/resources/hatoba/cluster/update.go:36]: Thanks!! Fixed!!

File: nifcloud/resources/hatoba/cluster/update.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	if d.HasChange("node_pools") {
```
[the remark this answers is not in the document]
Comment (aokumasan) [reply nifcloud/resources/hatoba/cluster/update.go:39]: tashikani... I fixed to add the new node pool before deleting old node pool 🚀

File: nifcloud/resources/hatoba/cluster/schema.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fuku2014) [reply nifcloud/resources/hatoba/cluster/schema.go:36]: thx!

File: nifcloud/resources/hatoba/cluster/update.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (fuku2014) [reply nifcloud/resources/hatoba/cluster/update.go:36]: Thanks!

File: nifcloud/resources/hatoba/cluster/flattener.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	if err := d.Set("network_config", flattenNetworkConfig(cluster.NetworkConfig)); err != nil {
+		return err
+	}
+
+	if err := d.Set("node_pools", flattenNodePools(cluster.NodePools)); err != nil {
+		return err
+	}
+
```
Comment (fuku2014) [inline nifcloud/resources/hatoba/cluster/flattener.go:50]: Would you please also set `locations` to state?

File: nifcloud/resources/hatoba/cluster/schema.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fuku2014) [reply nifcloud/resources/hatoba/cluster/schema.go]: 🚀 Thanks 🚀

File: nifcloud/resources/hatoba/cluster/update.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (fuku2014) [reply nifcloud/resources/hatoba/cluster/update.go:39]: uum... 
There is a problem that the name will be duplicated if it is added first 😇 

`Error: failed creating Hatoba cluster node pool: Client.InvalidParameterDuplicate.NodePoolName: The node pool 'default' already exists.`

[the remark this answers is not in the document]
Comment (aokumasan) [reply nifcloud/resources/hatoba/cluster/update.go:39]: oh...😇

I fixed node pool name and instance type schema to force new 😇

File: nifcloud/resources/hatoba/cluster/flattener.go (hunk 5 above)
Comment (aokumasan) [reply nifcloud/resources/hatoba/cluster/flattener.go:54]: Oh... I forgot 🙏

Fixed!!!

**Comment by fuku2014** [comment]
Yoshi!

**Review by fuku2014 — APPROVED** [review]

**Pull request merged**